--- a/hybris/common/jb/linker.c
+++ b/hybris/common/jb/linker.c
@@ -1284,14 +1284,15 @@ static int reloc_library(soinfo *si, Elf_Rel *rel, unsigned count)
         if(sym != 0) {
             sym_name = (char *)(strtab + symtab[sym].st_name);
             INFO("HYBRIS: '%s' checking hooks for sym '%s'\n", si->name, sym_name);
-            sym_addr = _get_hooked_symbol(sym_name, si->name);
-            if (sym_addr != NULL) {
+            sym_addr = (unsigned)(uintptr_t)
+                       _get_hooked_symbol(sym_name, si->name);
+            if (sym_addr != 0) {
                 INFO("HYBRIS: '%s' hooked symbol %s to %x\n", si->name,
                                                   sym_name, sym_addr);
             } else {
                s = _do_lookup(si, sym_name, &base);
             }
-            if(sym_addr == NULL)
+            if(sym_addr == 0)
             if(s == NULL) {
                 /* We only allow an undefined symbol if this is a weak
                    reference..   */
